extensions/wikia/Search/WikiaSearchController.class.php

```diff
@@ -663,43 +665,80 @@ protected function addRightRailModules( Wikia\Search\Config $searchConfig ) {
 
 		// SUS-1219: Use proper sanity check to handle space-only queries correctly
 		$hasTerms = $searchConfig->getQuery()->hasTerms();
-		if ( $wgEnableFandomStoriesOnSearchResultPage && $wgLang->getCode() === 'en' && $hasTerms ) {
+		if ( $wgEnableFandomStoriesOnSearchResultPage && $wgLang->getCode() === 'en' &&
+		     $hasTerms
+		) {
 			$query = $searchConfig->getQuery()->getSanitizedQuery();
 
-			$fandomStories = \WikiaDataAccess::cache(
-				wfSharedMemcKey( static::FANDOM_STORIES_MEMC_KEY, $query ),
-				\WikiaResponse::CACHE_STANDARD,
-				function() use ( $query ) {
-					return ( new \Wikia\Search\Services\FandomSearchService() )->query( $query );
-				}
-			);
+			if ( $wgEnableFandomStoriesSearchLogging ) {
+				$log = WikiaLogger::instance();
+				$log->info( __METHOD__ . ' - Querying Fandom Stories', [
+					'query' => $query,
+				] );
+			}
+
+			$searchCommand = $this->buildSearchCommand( $query );
```

Comment:
I'm objecting returning function type and assigning it to a variable. You could use that result directly and provide lambda function as callable instead. It would be easier to read and understand, and having a method invoked twice is not an issue.